### PR TITLE
Add containsAll, containsAny and doesNotContain in Collection Macros

### PR DESCRIPTION
**Pull request by michaelnabil230**
This PR to add `containsAll`,`containsAny` and `doesNotContain` in to Collection.
These functions allow checking the presence of any, none, or all values.

ِExamples for use:
```php

use Illuminate\Http\Request;

class UserController
{
    public function __invoke(Request $request)
    {
        $request->validate([
            'names' => ['required', 'array'],
        ]);

        $names = $request->collect('names');

        if ($names->containsAll(['Freek', 'Michael'])) {
            # code...
        }

        if ($names->containsAny(['Freek Van der Herten', 'Michael Nabil', 'Khaled Ahmed'])) {
            # code...
        }

        if ($names->doesNotContain(['Marian', 'Ahmed', 'Menna'])) {
            # code...
        }
    }
}
```

Another example:
```php 

use Illuminate\Http\Request;

class BookController
{
    public function __invoke(Request $request)
    {
        $books = ['book1', 'book2', 'book3'];
        
        if (collect($books)->containsAll(['book1', 'book2'])) {
            # code...
        }

        if (collect($books)->containsAny(['book1', 'book2', 'book5', 'book6'])) {
            # code...
        }

        if (collect($books)->doesNotContain(['book4', 'book5'])) {
            # code...
        }
    }
}


```

### CODE REVIEW

File: src/CollectionMacroServiceProvider.php [hunk 1]
```diff
@@ -68,6 +68,8 @@ private function macros(): array
             'validate' => \Spatie\CollectionMacros\Macros\Validate::class,
             'weightedRandom' => \Spatie\CollectionMacros\Macros\WeightedRandom::class,
             'withSize' => \Spatie\CollectionMacros\Macros\WithSize::class,
+            'containsAll' => \Spatie\CollectionMacros\Macros\ContainsAll::class,
```
Comment (freekmurze) [inline src/CollectionMacroServiceProvider.php:71]: Put this in alphabetical order

File: tests/Macros/ContainsAllTest.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+use Illuminate\Support\Collection;
+use Spatie\CollectionMacros\Test\TestCase;
+
+class ContainsAllTest extends TestCase
+{
+    /** @test */
+    public function it_contains_all()
+    {
+        $c = Collection::make([1, 3, 5]);
```
Comment (freekmurze) [inline tests/Macros/ContainsAllTest.php:13]: Don't use abbreviations, rename to `$collection`

File: tests/Macros/ContainsAnyTest.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+use Illuminate\Support\Collection;
+use Spatie\CollectionMacros\Test\TestCase;
+
+class ContainsAnyTest extends TestCase
+{
+    /** @test */
+    public function it_contains_any()
+    {
+        $c = Collection::make([1, 3, 5]);
```
Comment (freekmurze) [inline tests/Macros/ContainsAnyTest.php:13]: Don't use abbreviations.

**Review by freekmurze — CHANGES_REQUESTED** [review]
Please also add an example to the readme. Make sure it follows the style of the other examples.

**Comment by michaelnabil230** [comment]
@freekmurze Everything is good now

File: README.md [hunk 4]
```diff
@@ -66,6 +66,9 @@ The package will automatically register itself.
 - [`insertBefore`](#insertbefore)
 - [`insertBeforeKey`](#insertbeforekey)
 - [`none`](#none)
+- [`containsAll`](#containsAll)
```
Comment (freekmurze) [inline README.md:69]: Could you put these new ones after `collectBy`?

**Comment by michaelnabil230** [comment]
All is done now

**Comment by michaelnabil230** [comment]
@freekmurze  All is done now

File: README.md [hunk 5]
```diff
@@ -544,6 +547,33 @@ collect(['name' => 'foo'])->none(function ($key, $value) {
...
+### `containsAny`
+
+Checks whether a collection contains any values of the items that are identified.
+
+```php
+collect(['foo'])->containsAny(['foo', 'baz']); // returns true
+collect(['foo'])->containsAny(['bar', 'baz']); // returns false
+```
+
+### `containsAnyNone`
```
Comment (freekmurze) [inline README.md:568]: I think we can make this one better:

Could you rename this on to `doesNotContain`? 

These should all work:

- `doesNotContain($value)`
- `doesNotContain($value, $anotherValue)`
- `doesNotContain([$value, $anotherValue])` // does same as above

**Review by freekmurze — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (michaelnabil230) [reply README.md:568]: Do you want to rename `containsAnyNone ` to `doesNotContain` ?

Comment (freekmurze) [reply README.md:568]: Yup 👍 

I think that sounds more natural.

**Comment by michaelnabil230** [comment]
@freekmurze I did the instructions I mentioned, but I feel that something is missing or not well.

**Comment by michaelnabil230** [comment]
There is something else too

When I do unit test there is something that doesn't work in some way but in things that have nothing to do with my code. Can I view it here or create a issue?

**Comment by freekmurze** [comment]
Can you believe, I coded up the very same macro's as the ones provided in this PR (which I had forgotten about, sorry 😅)

I took some hints from this PR to polish my own functions.

So this won't be merged, but you'll be happy to know that these are now available in the latest release.

**Pull request closed without merging**